lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c

```diff
@@ -1051,17 +1061,16 @@ static PT_THREAD(peripheral_disconnect_task(struct pt *pt, pbio_task_t *task)) {
     if (peri->con_handle != NO_CONNECTION) {
         PT_WAIT_WHILE(pt, write_xfer_size);
         GAP_TerminateLinkReq(peri->con_handle, 0x13);
-        PT_WAIT_UNTIL(pt, hci_command_status);
+        PT_WAIT_UNTIL(pt, peri->con_handle == NO_CONNECTION);
```

Comment:
I would keep the waiting for command status. The disconnect could happen for a different reason other than the request made here, so the program flow would continue while the command is still being processed on the BT chip.

Reply:
So (in nominal conditions) are we guaranteed to get two events and always in that order?

Reply:
yes. and even if the disconnect event happens first due to non-nominal conditions `peri->con_handle == NO_CONNECTION` is still going to be true after the status event.